### PR TITLE
Update instance loop support for namespaced classes and plans

### PR DESCRIPTION
**Pull request by jmrodri**

### CODE REVIEW

**Comment by jmrodri** [comment]
I've signed it

**Comment by eriknelson** [comment]
Blocked by #2137, which explicitly allows namespaced ref updates to pass through the instance's references subresource.

File: .gitignore [hunk 1]
```diff
@@ -25,3 +25,4 @@ contrib/examples/consumer/Gopkg.lock
 contrib/examples/consumer
 contrib/examples/vendor/*
 integration.test*
+debug.test*
```
Comment (MHBauer) [inline .gitignore]: where is this coming from?

Comment (eriknelson) [reply .gitignore]: Was some kind of binary that showed up after running a failed make command, I can't remember which. I'm assuming we would prefer not to have that checked in.

Comment (carolynvs) [reply .gitignore]: Yeah I've seen those pop up too after running the tests.

**Pull request merged**